From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Tue, 17 Feb 2026 04:56:55 +0000
Subject: [PATCH] Gecko - IronFox Settings: WebGL

Provides support for a UI setting to enable/disable WebGL (including on a per-site basis), as a custom RFP target.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/dom/canvas/ClientWebGLContext.cpp b/dom/canvas/ClientWebGLContext.cpp
index 12492d606cc2..0fd90c1a8b19 100644
--- a/dom/canvas/ClientWebGLContext.cpp
+++ b/dom/canvas/ClientWebGLContext.cpp
@@ -883,6 +883,10 @@ bool ClientWebGLContext::CreateHostContext(const uvec2& requestedSize) {
         .options = options,
     };
 
+    if (ShouldResistFingerprinting(RFPTarget::DisableWebGL)) {
+      return Err("WebGL is currently disabled.");
+    }
+
     // -
 
     auto useOop = IsWebglOutOfProcessEnabled();
diff --git a/toolkit/components/resistfingerprinting/RFPTargets.inc b/toolkit/components/resistfingerprinting/RFPTargets.inc
index b77c1fd1f9ac..b87224e93d7e 100644
--- a/toolkit/components/resistfingerprinting/RFPTargets.inc
+++ b/toolkit/components/resistfingerprinting/RFPTargets.inc
@@ -113,6 +113,7 @@ ITEM_VALUE(WebGLVendorSanitize,         77)
 ITEM_VALUE(WebGLVendorConstant,         78)
 ITEM_VALUE(WebGLVendorRandomize,        79)
 ITEM_VALUE(WebGLRendererConstant,       80)
+ITEM_VALUE(DisableWebGL,81)
 
 // !!! Adding a new target? Rename PointerId and repurpose it.
 
-- 
